### PR TITLE
Locator API examples + validation of across frame navigations behavior

### PR DESCRIPTION
**Pull request by inancgumus**
This PR updates README for the Locator API and verifies the locator API behavior across frame navigations.
    
* Adds Locator API examples (`examples/`)
    * Locator page object model example
    * Locator across frame navigations example
* Updates the missing functionality table (README)
* Add Locator API examples (README)
    
The Locator API  works across frame navigations. The only place where it won't work is if the underlying frame somehow gets changed (rare?).

### CODE REVIEW

**Review by ankur22 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -311,6 +311,139 @@ export default function() {
...
+
+Here's an example to better understand the differences.
+
+```js
+import launcher from "k6/x/browser";
+
+export default function () {
+  const browser = launcher.launch('chromium', {
+    headless: __ENV.XK6_HEADLESS ? true : false,
+  });
```
Comment (inancgumus) [inline README.md:330]: Ah, I forgot to remove these...

File: README.md [hunk 2]
```diff
@@ -311,6 +311,139 @@ export default function() {
...
+    this.page.waitForNavigation();
+  }
+
+  current() {
+    return this.currentBet.innerText();
+  }
+}
+
+export default function () {
+  const browser = launcher.launch('chromium', { headless: __ENV.XK6_HEADLESS ? true : false });
```
Comment (inancgumus) [inline README.md:423]: Also, here... 🤦

File: examples/locator.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  If you were to query the elements once and save them as
+  below, you would see an error after the initial navigation.
+  Try it!
+
+    const head = page.$("input[value='Bet on heads!']");
+    const tail = page.$("input[value='Bet on tails!']");
+
+  The Locator API allows you to get a fresh element handle each
+  time you use one of the locator methods. And, you can carry a
+  locator across frames. Let's create two locators each locates
```
Comment (inancgumus) [inline examples/locator.js:26]: "Across frame navigations"

**Review by inancgumus — COMMENTED** [review]
TODO

File: README.md [hunk 4]
```diff
@@ -311,6 +311,139 @@ export default function() {
...
+  });
+  const context = browser.newContext();
+  const page = context.newPage();
+  page.goto("https://test.k6.io/flip_coin.php", {
+    waitUntil: "networkidle",
+  });
+
+  /*
+  In this example, we will use two locators, matching a
+  different betting button (head and tail) on the page.
```
Comment (imiric) [inline README.md:339]: Nit: the values are actually "heads" and "tails", i.e. it's never used as singular.

You might want to change the variable names as well.

File: README.md [hunk 5]
```diff
@@ -311,6 +311,139 @@ export default function() {
...
+  console.log(currentBet.innerText());
+
+  page.close();
+  browser.close();
+}
+```
+
+The Locator API can help you abstract a page to simplify testing it. To
+do that, you can use a pattern called the Page Object Model. Here's an
+example.
```
Comment (imiric) [inline README.md:384]: I'm not sure if we should include this example in the README. It's a good example and we should have it in the `examples` directory, I just think it's redundant to have two Locators API examples in a README already overloaded with examples.

File: examples/locator.js (hunk 3 above)
Comment (imiric) [reply examples/locator.js:26]: Also, let's make sure the files in the `examples` directory are synced with the ones in the README (minus the `headless` conditional), after addressing any feedback.

**Review by imiric — CHANGES_REQUESTED** [review]
LGTM, just minor things.

Both are good examples, and we could include the non-POM one in the [v0.4.0 release notes](https://github.com/grafana/xk6-browser/pull/418#discussion_r908540093).

File: README.md (hunk 5 above)
Comment (inancgumus) [reply README.md:384]: What about linking to it from the README?

**Comment by inancgumus** [comment]
> LGTM, just minor things.
> 
> Both are good examples, and we could include the non-POM one in the [v0.4.0 release notes](https://github.com/grafana/xk6-browser/pull/418#discussion_r908540093).

Thanks! I made all the changes you requested. The only difference is that the README links to the POM example.

**Review by imiric — APPROVED** [review]

**Pull request merged**